python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -331,22 +309,25 @@ def __getitem__(self, this_id):
             this_id (int): This is the key of an object in the object dictionary
         """
         logger.debug("__getitem__")
+        # Is this an Incomplete Object?
+        if this_id in self._incomplete_objects:
+            return IncompleteObject(self, this_id)
+
+        # Nope, so try to find it and raise an exception if not
         try:
             return self._objects[this_id]
         except KeyError as err:
             logger.debug("Repo KeyError: %s" % err)
             logger.debug("Keys: %s id: %s" % (self._objects.keys(), this_id))
-            if this_id in self._incomplete_objects:
```

Comment:
@drmarkwslater OK I think I'm seeing this now. I think the object is never loaded here in this case?

Apologies I think I'm getting confused from trying to debug another issue with loading of jobs. I forgot it's elsewhere the object is actually loaded and so moves between the lists. This should be safe in that case if it's locked so my apologies again.
(I saw this happen a few times in the past but now I realize that was likely a threading issue as it was before we locked some transactions).